detections/application/splunk_command_and_scripting_interpreter_detect_risky_spl_mltk.yml

```diff
@@ -0,0 +1,60 @@
+name: Splunk Command and Scripting Interpreter Detect Risky SPL MLTK
+id: b4aefb5f-1037-410d-a149-1e091288ba33
+version: 1
+date: '2022-06-16'
+author: Abhinav Mishra, Kumar Sharad, Namratha Sreekanta and Xiao Lin, Splunk
+type: Anomaly
+datamodel:
+- Splunk_Audit
+description: 'The following analytic uses a pre-trained machine learning text classifier to detect potentially risky commands. A command is deemed risky based on the presence of certain trigger keywords, along with the context and the role of the user (please see references). The model takes as input the command text, user and search type and outputs a risk score between [0,1]. A high score indicates higher likelihood of a command being risky.'
+search: '| tstats `security_content_summariesonly` count min(_time) as firstTime max(_time) as lastTime from datamodel=Splunk_Audit.Search_Activity where Search_Activity.search_type=adhoc Search_Activity.user!=splunk-system-user by Search_Activity.search Search_Activity.user Search_Activity.search_type
+  | eval spl_text = 'Search_Activity.search'. " " .'Search_Activity.user'. " " .'Search_Activity.search_type'
+  | dedup spl_text
+  | apply DetectRiskySPL
+  | where risk_score > 0.6
+  | table Search_Activity.search, Search_Activity.user, Search_Activity.search_type, risk_score
+  | `drop_dm_object_name(Search_Activity)`
+  |`splunk_command_and_scripting_interpreter_detect_risky_spl_mltk_filter`'
+how_to_implement: This detection depends on the MLTK app which can be found here - https://splunkbase.splunk.com/app/2890/ and the Splunk Audit datamodel which can be found here - https://splunkbase.splunk.com/app/1621/. Additionally, you need to be ingesting logs which include Search_Activity.search, Search_Activity.user, Search_Activity.search_type from your endpoints.
```

Comment:
Minor thing here, but might be worth mentioning changing the risk_score threshold based on their environments.

Reply:
Updated.